lib/components/segmented-control/segment.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import clsx from "clsx";
+import { Text } from "../Typography";
+import * as quillIcons from "@deriv/quill-icons";
+import { SegmentedControlProps } from "./base";
+
+interface SegmentProps {
+    className?: string;
+    icon?: string | React.ReactNode;
+    isDisabled?: boolean;
+    isSelected?: boolean;
+    label?: string;
+    onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
+    size?: SegmentedControlProps["size"];
+}
+
+export const Segment = ({
+    className,
+    icon,
+    isDisabled,
+    isSelected,
+    label,
+    onClick,
+    size,
+}: SegmentProps) => {
+    const QuillIcon =
+        typeof icon === "string" &&
+        (quillIcons[icon as keyof typeof quillIcons] as React.ElementType);
```

Comment:
I love the idea of allowing consumers to pass string for icons to leverage on using the Quill Icons, however when might need to check if this will impact the bundle size of our library since we are importing the whole quill icons here.

Reply:
ok, i'll check )

Reply:
good catch, thank you! with this approach, the bundle looks terrifying:
<img width="845" alt="Screenshot 2024-03-27 at 4 30 09 PM" src="https://github.com/deriv-com/quill-ui/assets/103177211/48f738e9-39a4-4bda-aac7-0850f99c61bd">
it should be built in 1.30s and not be that huge.
I'll get rid of this approach, which I had to use because of [this known issue with storybook Documentation](https://github.com/storybookjs/storybook/issues/17098), which gets stuck in an infinite loop when you try passing an icon as a component from within a story.
This will be a hard nut to crack.

Reply:
There must be a certain configuration in the storybook to avoid that, thanks for checking the bundle size awesome!

Reply:
@prince-deriv i made my component support 'placeholder' as 'icon' value :) It won't hurt